src/Blueprint/Blueprint/Cmdlets/NewAzureRMBlueprintAssignment.cs

```diff
@@ -0,0 +1,90 @@
+﻿using Microsoft.Azure.Commands.Blueprint.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.Blueprint.Models;
+using System;
+using System.Collections;
+using System.Linq;
+using System.Management.Automation;
+using System.Management.Automation.Language;
+using Microsoft.Azure.PowerShell.Cmdlets.Blueprint.Properties;
+using ParameterSetNames = Microsoft.Azure.Commands.Blueprint.Common.BlueprintConstants.ParameterSetNames;
+using ParameterHelpMessages = Microsoft.Azure.Commands.Blueprint.Common.BlueprintConstants.ParameterHelpMessages;
+using System.Text.RegularExpressions;
+using Microsoft.Azure.Commands.Blueprint.Common;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.Commands.Blueprint.Cmdlets
+{
+    [Cmdlet(VerbsCommon.New, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "BlueprintAssignment", SupportsShouldProcess = true)]
+    public class NewAzureRmBlueprintAssignment : BlueprintCmdletBase
+    {
+        #region Parameters
+        [Parameter(ParameterSetName = ParameterSetNames.CreateBlueprintAssignment, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = ParameterHelpMessages.BlueprintAssignmentName)]
+        [ValidatePattern("^[0-9a-zA-Z_-]*$", Options = RegexOptions.Compiled | RegexOptions.CultureInvariant)]
```

Comment:
>] [](start = 39, length = 1)

We can probably add a length validator as well or add a quantifier to the regex.

Reply:
I can add length validator no problem. But I checked this on the Portal and it didn't look like we had a length requirement for assignment name(I was able to enter arbitrarily long assignment name there). Do we have a check for that during deployment?

---
In reply to: [257119662](https://github.com/Azure/azure-powershell/pull/8537#discussion_r257119662) [](ancestors = 257119662)

Reply:
UI might not check - I'd check if backend does some validation.

---
In reply to: [257330223](https://github.com/Azure/azure-powershell/pull/8537#discussion_r257330223) [](ancestors = 257330223,257119662)

Reply:
I couldn't find any length validation on the backend either. 

---
In reply to: [257382671](https://github.com/Azure/azure-powershell/pull/8537#discussion_r257382671) [](ancestors = 257382671,257330223,257119662)

Reply:
Then never mind :)

---
In reply to: [257712941](https://github.com/Azure/azure-powershell/pull/8537#discussion_r257712941) [](ancestors = 257712941,257382671,257330223,257119662)